File: tests/Generator/resource/swift/complex/ReferencePropertyType.swift
// Represents a reference to a definition type
class ReferencePropertyType: PropertyType {
    var target: String?
    var template: Dictionary<String, String>?
    var _type: String? = "reference"

    enum CodingKeys: String, CodingKey {
        case target = "target"
        case template = "template"
        case _type = "type"
    }
}

File: tests/Generator/resource/csharp/complex/ReferencePropertyType.cs
using System.Text.Json.Serialization;

/// <summary>
/// Represents a reference to a definition type
/// </summary>
public class ReferencePropertyType : PropertyType
{
    [JsonPropertyName("target")]
    public string? Target { get; set; }

    [JsonPropertyName("template")]
    public System.Collections.Generic.Dictionary<string, string>? Template { get; set; }

}

